build.sbt

```diff
@@ -481,6 +489,106 @@ lazy val unidocSettings =
       (JavaUnidoc / unidoc / unidocAllSources).value.map(apiDocSourceFilter)
   )
 
+lazy val genTunablesDocSettings = Seq(
+  Compile / genTunablesDoc := {
+    val stream = (propgen / streams).value
+    val dafExtFile =
+      (propgen / Compile / resources).value.find(_.getName == "dafext.xsd").get
+    val outputDocFile = (Compile / target).value / "tunables.md"
+    // parse xsd file
+    val dafExtXml = scala.xml.XML.loadFile(dafExtFile)
+    // extract tunables information
+    val tunablesElements =
+      (dafExtXml \ "element").filter(_ \@ "name" == "tunables") \\ "all" \ "element"
+    // build documentation
+    val documentationTuple = tunablesElements.map { ele =>
+      val subtitle = ele \@ "name"
+      val documentation = (ele \ "annotation" \ "documentation").text.trim.replaceAll(" +", " ")
+      val default = ele \@ "default"
+      (subtitle, documentation, default)
+    }
+    val (deprecated, nonDeprecated) = documentationTuple.partition { t =>
+      t._2.startsWith("Deprecated")
+    }
+    if (nonDeprecated.isEmpty) {
+      sys.error("tunables document generation failed as non-deprecated elements list is empty")
```

Comment:
This just outputs a log message but doesn't actually fail the task. I've found the best way to create a failure with an error message is to replace `sys.error` with this:
```scala
throw new MessageOnlyException("error message here")
```
It also means you can move the following code out of the else block.